lib/sles4sap/sap_deployment_automation_framework/configure_workload_tfvars.pm

```diff
@@ -0,0 +1,393 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::configure_workload_tfvars;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use utils qw(write_sut_file);
+use sles4sap::sap_deployment_automation_framework::deployment qw(get_os_variable get_fencing_mechanism);
+use sles4sap::sap_deployment_automation_framework::naming_conventions
+  qw(convert_region_to_short generate_resource_group_name);
+use sles4sap::sap_deployment_automation_framework::deployment_connector qw(find_deployment_id no_cleanup_tag);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation that help with preparation of
+'WORKLOAD-ZONE' tfvars file.
+Generated file is following example template:
+https://github.com/Azure/SAP-automation-samples/blob/main/Terraform/WORKSPACES/SYSTEM/LAB-SECE-SAP04-L00/LAB-SECE-SAP04-L00.tfvars
+
+=cut
+
+our @EXPORT = qw(
+  create_workload_tfvars
+  compile_tfvars_section
+  write_tfvars_file
+);
+
+=head2 write_tfvars_file
+
+    write_tfvars_file(tfvars_file=>'/path/towards/happiness', tfvars_data=>$tfvars_data);
+
+Data provided in B<$args{tfvars_data}> is compiled into tfvars file format and written into target B<$args{tfvars_file}>
+file located on deployer VM.
+B<$args{tfvars_data}> is a HASHREF containing individual sections that should be included in final tfvars content.
+B<file_header> is placed as a comment as a first line of the defined section.
+
+B<Example:>
```

Comment:
could you please extend the example with the expected generated tfvar file?

Reply:
Or maybe extend the one in compile_tfvars_section... I do not understand exactly what is done by this function and what is done by the other `compile_tfvars_section` and why they are separated.

Reply:
It does two things:
1. converts data from hash to tfvars format `key="value"` ... basically a multiline string
2. ensures that header stays on top of each section. If you iterate over hash keys you will get unordered array and header can appear anywhere.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21611/files#diff-c651ed81175728acdeb6fdccf551a27ccb0856cffc5577e6fcb4d62a27cbcd1eR110

Reply:
Example added